src/engagement_db_coda_sync/lib.py

```diff
@@ -297,6 +297,11 @@ def _update_engagement_db_message_from_coda_message(engagement_db, engagement_db
                 raise e
             correct_dataset = coda_config.default_ws_dataset
 
+        # Ensure the message isn't being WS-corrected to the dataset it's already in.
+        assert correct_dataset != engagement_db_message.dataset, \
+            f"Engagement db message '{engagement_db_message.message_id}' (text '{engagement_db_message.text}') " \
+            f"is being WS-corrected to dataset '{correct_dataset}', but is currently in this dataset already."
+
```

Comment:
add a todo to add a solution to the issue once we have studied it.

Reply:
agreed, added